txnsync/bitmask.go

```diff
@@ -71,13 +70,16 @@ func (b *bitmask) trimBitmask(entries int) {
 		bestSize = (entries-numExists)*2 + 1
 	}
 	switch bitmaskType {
+	case 0:
+		*b = (*b)[:bestSize]
 	case 1:
 		(*b)[0] = 1
 		for i := range *b {
 			if i != 0 {
 				(*b)[i] = 255 - (*b)[i] // invert bits
 			}
 		}
+		*b = (*b)[:bestSize]
```

Comment:
now that you're writing it like this, it looks like we can modify the above for loop to be
```golang
for i := 1; i < bestSize; i++ {
   (*b)[i] = 255 - (*b)[i] // invert bits
}
```

but we can also wait with this for a separate PR, since this one looks good to me.